apollo-runtime/src/main/java/com/apollographql/android/cache/normalized/lru/LruCacheStore.java

```diff
@@ -0,0 +1,87 @@
+package com.apollographql.android.cache.normalized.lru;
+
+import android.app.ActivityManager;
+import android.content.Context;
+
+import com.apollographql.android.api.graphql.internal.Optional;
+import com.apollographql.android.cache.normalized.CacheStore;
+import com.apollographql.android.cache.normalized.Record;
+import com.nytimes.android.external.cache.Cache;
+import com.nytimes.android.external.cache.CacheBuilder;
+import com.nytimes.android.external.cache.Weigher;
+
+import java.util.Collections;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import static android.content.Context.ACTIVITY_SERVICE;
+import static android.content.pm.ApplicationInfo.FLAG_LARGE_HEAP;
+
+public final class LruCacheStore extends CacheStore {
+
+  private final Cache<String, Record> lruCache;
+
+  public static LruCacheStore create(Context context) {
+    return new LruCacheStore(Optional.of(calculateMemoryCacheSize(context)), Optional.<Long>absent());
+  }
+
+  public static LruCacheStore createWithMaximumEntrySize(long entrySize) {
+    return new LruCacheStore(Optional.<Long>absent(), Optional.of(entrySize));
+  }
+
+  public static LruCacheStore createWithMaximumByteSize(long sizeInBytes) {
+    return new LruCacheStore(Optional.of(sizeInBytes), Optional.<Long>absent());
+  }
+
+  public static LruCacheStore createUnbounded() {
+    return new LruCacheStore(Optional.<Long>absent(), Optional.<Long>absent());
+  }
+
+  /**
+   * Derived from https://github.com/square/picasso cache sizing.
+   */
+  private static long calculateMemoryCacheSize(Context context) {
+    ActivityManager activityManager = (ActivityManager) context.getSystemService(ACTIVITY_SERVICE);
```

Comment:
I don't think so -- but we depend on android for the SQL store, so this could be moved to the "Android cache" module, and leave the rest of the LRU stuff in a "core cache" module. Though, we are starting to get tons of modules :)